programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -287,6 +322,45 @@ pub mod gateway {
 
         Ok(())
     }
+
+    pub fn execute(
+        ctx: Context<Execute>,
+        sender: Pubkey,
+        data: Vec<u8>,
+    ) -> Result<()> {
+        let pda = &mut ctx.accounts.pda;
+        require!(!pda.deposit_paused, Errors::DepositPaused);
+
+        // NOTE: have to manually create Instruction, pack it and invoke since there is no crate for contract
+        // since any contract with on_call instruction can be called
+        let instruction_data = CallableInstruction::OnCall {
+            sender,
+            data,
+        }
+        .pack();
+
+        // NOTE: calling function in other program without passing accounts seems very limitting in what can be done
+        // every account that instruction interacts with has to be predetermined and set before the call, and various callable contracts might have different behavior and need different accounts
```

Comment:
Do you think it is a set of data that could be potentially represented during the call on ZetaChain using a standard?

A bit like the idea of the Bitcoin memo standard.

User provide a generate bytes input, following a standard, this input is decoded to represent the full call on Solana.
This could be a generic field that we would need to other chain, it seems there will be lot of considerations to have as well for TON smart contract calls

Reply:
> Also important, program can not access those accounts when it executes transaction, the only way is that account need to be passed with instruction, basically preloaded before execution happens.

Yes, but this is a piece of information that could be passed in the gateway on ZetaChain?